### PR TITLE
feat: Add support for new float syntax

### PR DESCRIPTION
**Pull request by spotandjake**
I accidentally closed my other pr, when I deleted my fork, reopening from my new fork.

Closes: #154

### CODE REVIEW

File: editor-extensions/vscode/syntaxes/grain.json [hunk 1]
```diff
@@ -838,6 +838,15 @@
             }
           }
         },
+        {
+          "match": "\\b(0[xX][\\da-fA-F]+)(\\.[\\da-fA-F]*)?([p`P][+-]?\\d[\\d_]*)?([fdwW]?)\\b",
```
[the remark this answers is not in the document]
Comment (ospencer) [reply editor-extensions/vscode/syntaxes/grain.json:842]: Also looks like underscores aren't accepted in the mantissa.

[the remark this answers is not in the document]
Comment (spotandjake) [reply editor-extensions/vscode/syntaxes/grain.json:842]: I think we accept it in our implementation. 
image

[the remark this answers is not in the document]
Comment (ospencer) [reply editor-extensions/vscode/syntaxes/grain.json]: Yes, I meant that the underscores weren't accepted by your regex here. I'll add them.

File: editor-extensions/vscode/syntaxes/grain.json [hunk 2]
```diff
@@ -838,6 +838,15 @@
             }
           }
         },
+        {
+          "match": "\\b(0[xX][\\da-fA-F]+)(\\.[\\da-fA-F]*)?([pP][+-]?\\d[\\d_]*)?([fdwW]?)\\b",
```
Comment (ospencer) [inline editor-extensions/vscode/syntaxes/grain.json:842]:
```suggestion
          "match": "\\b(0[xX][\\da-fA-F][\\da-fA-F_]*)(\\.[\\da-fA-F][\\da-fA-F_]*)?([pP][+-]?\\d[\\d_]*)?([fdwW]?)\\b",
```

**Comment by ospencer** [comment]
image

**Review by ospencer — APPROVED** [review]

**Pull request merged**